composeml/label_maker.py

```diff
@@ -101,21 +102,31 @@ def df_to_labels(df):
                     labels[cutoff_time] = label
 
                 cutoff_time = offset_time(df.index, gap)
+                not_done = progress_bar.n < progress_bar.total
```

Comment:
why does this get called if the progress bar is already completed?

also, i think it's easier to read if we do

```
if progress_bar.n < progress_bar.total:
    progress_bar.update(n=1)
```

Reply:
I thought the same, but with these adjustments:
```
-                not_done = progress_bar.n < progress_bar.total
-                progress_bar.update(n=int(not_done))
+                progress_bar.update(n=1)
```
Not sure why the progress bar renders this way.
```
150it [00:00, 316.63it/s]                                                                                 
```

Reply:
why would `progress_bar.n < progress_bar.total` ever be false?

Reply:
After a closer look, it seems like `GroupBy.apply` applies `df_to_labels` twice on the first group.

```
Elapsed: 00:00 | Remaining: ? | Progress:   0%|                      | customer_id: 0/125 

                     customer_id  amount
transaction_time                        
2014-01-01 03:13:51            1  120.52
2014-01-02 17:46:20            1   65.11
```

```
Elapsed: 00:00 | Remaining: 00:00 | Progress:  12%|█████▌           | customer_id: 15/125 

                     customer_id  amount
transaction_time                        
2014-01-01 03:13:51            1  120.52
2014-01-02 17:46:20            1   65.11

```
```
Elapsed: 00:00 | Remaining: 00:00 | Progress:  26%|███████████▊     | customer_id: 32/125 

                     customer_id  amount
transaction_time                        
2014-01-02 17:12:39            2  105.15
2014-01-04 05:45:43            2   51.95
```

As an alternative, maybe we can iterate the groups using a for loop?

Reply:
ya, let's iterate over the groups then.

one note, I think they changed this in pandas `0.25.0`. even if it does work with the latest, i think we're better off having a lower minimum version. https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.25.0.html#groupby-apply-on-dataframe-evaluates-first-group-only-once

Reply:
Made update to [iterate groups](https://github.com/FeatureLabs/compose-ml/pull/35/files#diff-22bf41f653d21f0ca3dd15dadf5e01e9R122).